src/dali_executor/dali_executor.cc

```diff
@@ -65,19 +65,12 @@ IDescr DaliExecutor::ScheduleInputCopy(const IDescr& input) {
   auto descriptor = buffer->get_descr();
   char* dst = reinterpret_cast<char*>(descriptor.data);
   for (auto& buf : input.buffers) {
-    thread_pool_.AddWork(
-        [descriptor, dst, buf](int) {
-          MemCopy(descriptor.device, dst, buf.device, buf.data, buf.size);
-        },
-        buf.size, true);
+    MemCopy(descriptor.device, dst, buf.device, buf.data, buf.size);
```

Comment:
Can't you just adjust thread_pool_ contraction and instead of `pipeline_(std::move(pipeline)), thread_pool_(GetNumThreads(), pipeline_.DeviceId(), false) {}` pass -1 as the id when working in the CPU only mode?
DALI does that.

Reply:
I tried passing `-1` (ref. `DetermineDeviceId()`), but it still failed with "failed to open `libnvidia-ml.so`". What I see, is that DALI passes `-99999` rather than `-1` in this case. I'll try it out, however it made me thinking, that tightly coupling `dali_backend` code with DALI might length the bugfix cycle, if there's any fix required in DALI code base. I'll discuss it with @banasraf